csharp/src/Ice/Proxy.cs

```diff
@@ -497,6 +523,10 @@ public static class Proxy
                 }
                 finally
                 {
+                    if (retryCount > 0)
+                    {
+                        reference.Communicator.DecRetryBufferSize(request.Size);
+                    }
```

Comment:
It's debatable whether or not we should wait for the response to decrease the retry buffer size. This could end up preventing many retries if the response takes a long time to receive. Another possibility could be to decrease the buffer size once the request is sent over a new connection or once a new connection is established?

Reply:
I don't think this will be a problem, I expect this to be set to a value high enough to allow normal request retry ratio,  this will kick in when there are many concurrent retries. I guess we should probably increase the default 1MB might be too low.

Reply:
Updated the default value to 100MB